gradle/libs.versions.toml

```diff
@@ -0,0 +1,153 @@
+flink116-metrics-dropwizard = { module = "org.apache.flink:flink-metrics-dropwizard", version.ref = "flink116" }
+flink116-streaming-java = { module = "org.apache.flink:flink-streaming-java", version.ref = "flink116" }
+flink116-table-api-java-bridge = { module = "org.apache.flink:flink-table-api-java-bridge", version.ref = "flink116" }
+flink117-avro = { module = "org.apache.flink:flink-avro", version.ref = "flink117" }
+flink117-connector-base = { module = "org.apache.flink:flink-connector-base", version.ref = "flink117" }
+flink117-connector-files = { module = "org.apache.flink:flink-connector-files", version.ref = "flink117" }
+flink117-metrics-dropwizard = { module = "org.apache.flink:flink-metrics-dropwizard", version.ref = "flink117" }
+flink117-streaming-java = { module = "org.apache.flink:flink-streaming-java", version.ref = "flink117" }
+flink117-table-api-java-bridge = { module = "org.apache.flink:flink-table-api-java-bridge", version.ref = "flink117" }
+google-libraries-bom = "com.google.cloud:libraries-bom:24.1.0"
+guava-guava = { module = "com.google.guava:guava", version.ref = "guava" }
+hadoop2-client = { module = "org.apache.hadoop:hadoop-client", version.ref = "hadoop2" }
+hadoop2-common = { module = "org.apache.hadoop:hadoop-common", version.ref = "hadoop2" }
+hadoop2-hdfs = { module = "org.apache.hadoop:hadoop-hdfs", version.ref = "hadoop2" }
+hadoop2-mapreduce-client-core = { module = "org.apache.hadoop:hadoop-mapreduce-client-core", version.ref = "hadoop2" }
+hadoop2-minicluster = { module = "org.apache.hadoop:hadoop-minicluster", version.ref = "hadoop2" }
+hadoop3-client = "org.apache.hadoop:hadoop-client:3.3.6"
+hive2-exec = { module = "org.apache.hive:hive-exec", version.ref = "hive2" }
+hive2-metastore = { module = "org.apache.hive:hive-metastore", version.ref = "hive2" }
+hive2-serde = { module = "org.apache.hive:hive-serde", version.ref = "hive2" }
+hive2-service = { module = "org.apache.hive:hive-service", version.ref = "hive2" }
+hive3-exec = { module = "org.apache.hive:hive-exec", version.ref = "hive3" }
+hive3-metastore = { module = "org.apache.hive:hive-metastore", version.ref = "hive3" }
+hive3-serde = { module = "org.apache.hive:hive-serde", version.ref = "hive3" }
+hive3-service = { module = "org.apache.hive:hive-service", version.ref = "hive3" }
+httpcomponents-httpclient5 = "org.apache.httpcomponents.client5:httpclient5:5.2.1"
+immutables-value = "org.immutables:value:2.9.2"
+jackson-bom = { module = "com.fasterxml.jackson:jackson-bom", version = "2.14.2" }
+jackson211-bom = { module = "com.fasterxml.jackson:jackson-bom", version = { strictly = "[2.11, 2.12[", prefer = "2.11.4"} }
+jackson212-bom = { module = "com.fasterxml.jackson:jackson-bom", version = { strictly = "[2.12, 2.13[", prefer = "2.12.3"} }
+jackson213-bom = { module = "com.fasterxml.jackson:jackson-bom", version =  { strictly = "[2.13, 2.14[", prefer = "2.13.4"} }
```

Comment:
I believe it's better to generally specify the particular version to be used in this whole file rather than having version ranges

Reply:
For places like this(where a rich version declaration with  strictly & prefer is used) and similar others I will put a single comments: please read a [head description to PR](https://github.com/apache/iceberg/pull/7694#issue-1723184954), I've updated it with a point about rich versions. Let me know if you have questions and/or suggestions.